agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/AiSampler.java

```diff
@@ -138,13 +146,23 @@ public String getDescription() {
   private static class RecordAndSampleWithItemCount implements SamplingResult {
 
     private final Attributes attributes;
+    @Nullable private final QuickPulse quickPulse;
 
-    RecordAndSampleWithItemCount(long itemCount) {
+    RecordAndSampleWithItemCount(long itemCount, @Nullable QuickPulse quickPulse) {
```

Comment:
instead of passing in `quickPulse` here, it may be better to use quickpulse to get the decision and pass the decision in here instead (that way this object is a nice simple immutable object)